notifikasjon/src/test/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/notifikasjon/river/OpprettSelvbestemtSakRiverTest.kt

```diff
@@ -0,0 +1,182 @@
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import no.nav.helse.rapids_rivers.testsupport.TestRapid
+import no.nav.helsearbeidsgiver.arbeidsgivernotifikasjon.ArbeidsgiverNotifikasjonKlient
+import no.nav.helsearbeidsgiver.arbeidsgivernotifkasjon.graphql.generated.enums.SaksStatus
+import no.nav.helsearbeidsgiver.domene.inntektsmelding.v1.Inntektsmelding
+import no.nav.helsearbeidsgiver.felles.BehovType
+import no.nav.helsearbeidsgiver.felles.EventName
+import no.nav.helsearbeidsgiver.felles.Key
+import no.nav.helsearbeidsgiver.felles.json.toJson
+import no.nav.helsearbeidsgiver.felles.json.toMap
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.model.Fail
+import no.nav.helsearbeidsgiver.felles.test.mock.mockInntektsmeldingV1
+import no.nav.helsearbeidsgiver.felles.test.rapidsrivers.firstMessage
+import no.nav.helsearbeidsgiver.felles.test.rapidsrivers.sendJson
+import no.nav.helsearbeidsgiver.inntektsmelding.notifikasjon.db.SelvbestemtRepo
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import java.util.UUID
+
+class OpprettSelvbestemtSakRiverTest : FunSpec({
+
+    val testRapid = TestRapid()
+    val mockUrl = "selvbestemt-lenke"
+    val mockSelvbestemtRepo = mockk<SelvbestemtRepo>()
+    val mockagNotifikasjonKlient = mockk<ArbeidsgiverNotifikasjonKlient>()
+
+    OpprettSelvbestemtSakRiver(mockUrl, mockSelvbestemtRepo, mockagNotifikasjonKlient).connect(testRapid)
+
+    beforeTest {
+        testRapid.reset()
+        clearAllMocks()
+    }
+
+    test("opprett sak") {
+        val sakId = UUID.randomUUID().toString()
+        val innkommendeMelding = innkommendeMelding()
+
+        coEvery { mockagNotifikasjonKlient.opprettNySak(any(), any(), any(), any(), any(), any(), any(), any()) } returns sakId
+        every { mockSelvbestemtRepo.lagreSakId(any(), any()) } returns 1
+
+        testRapid.sendJson(innkommendeMelding.toMap())
+
+        testRapid.inspektør.size shouldBeExactly 1
+
+        testRapid.firstMessage().toMap() shouldContainExactly mapOf(
+            Key.EVENT_NAME to innkommendeMelding.eventName.toJson(),
+            Key.UUID to innkommendeMelding.transaksjonId.toJson(),
+            Key.DATA to "".toJson(),
+            Key.SAK_ID to sakId.toJson()
+        )
+
+        coVerifySequence {
+            mockagNotifikasjonKlient.opprettNySak(
+                virksomhetsnummer = innkommendeMelding.inntektsmelding.avsender.orgnr,
+                merkelapp = "Inntektsmelding",
+                grupperingsid = innkommendeMelding.inntektsmelding.type.id.toString(),
+                lenke = "$mockUrl/im-dialog/kvittering/agi/${innkommendeMelding.inntektsmelding.type.id}",
+                tittel = "Inntektsmelding for ${innkommendeMelding.inntektsmelding.sykmeldt.navn}: " +
+                    "f. ${innkommendeMelding.inntektsmelding.sykmeldt.fnr.take(6)}",
+                statusTekst = "NAV trenger inntektsmelding",
```

Comment:
Er forsåvidt ikke relatert til denne commiten / testen, men tror vi må endre opprettSak for selvbestemt til å ta med en annen statusTekst (NAV trenger jo ikke inntektsmelding, ser ganske rart ut i oversikten med den overskriften..), evt ferdigstille saken også..

Reply:
Ny statustekst er nok ikke så dumt. Jeg kan høre med Daniel om han har tanker.
Ellers så ferdigstilles saken i dag, via feltet `initiellStatus = SaksStatus.FERDIG`.

Reply:
Kontra: Vi trenger vel ikke høre med Daniel. De andre sakene får statustekst "Mottatt - Se kvittering eller korriger inntektsmelding" ved lukking. Kan bruke den samme her.